packages/profile-sync-controller/src/controllers/authentication/AuthenticationController.ts

```diff
@@ -242,15 +264,22 @@ export default class AuthenticationController extends BaseController<
     }
   }
 
-  public async performSignIn(): Promise<string> {
+  public async performSignIn(): Promise<string[]> {
     this.#assertIsUnlocked('performSignIn');
-    return await this.#auth.getAccessToken();
+
+    const allPublicKeys = await this.#snapGetAllPublicKeys();
+
+    return await Promise.all(
+      allPublicKeys.map(async ([entropySourceId]) => {
+        return await this.#auth.getAccessToken(entropySourceId);
+      }),
+    );
   }
 
   public performSignOut(): void {
     this.update((state) => {
       state.isSignedIn = false;
-      state.sessionData = undefined;
+      state.srpSessionData = undefined;
```

Comment:
Is this maybe a bit safer?
```suggestion
      state.srpSessionData = {};
```

Reply:
Hmmm I don't know, because I don't want to end up in a knot in clients transforms and migrations.
The reason is that `srpSessionData` initializes as undefined, as should all optional state properties be, and to me it adds some overhead to check if `srpSessionData` is existing if it can also be an empty object.

I.e `Boolean({}) === true`, but we still consider srpSessionData to be empty in our logic.

WDYT?

Reply:
It's ok as long as we always treat it as potentially undefined.